### PR TITLE
feat: ts-monitor supports https and auth

### LINKED ISSUE #277: ts-monitor supports https and auth to report metrics
### Describe your question

ts-monitor needs to support the following functions:

a. Support https reporting
b. Support auth

### PR DESCRIPTION
**Pull request by yehong-z**
### What problem does this PR solve?

close #277 

### What is changed and how it works?

ts-monitor supports https and auth

### How Has This Been Tested?

```
[report]
  username = "admin"
  password = "pass"
  https-enable = true
```
Add username and password fields to the report configuration of ts-monitor, which can be used to authenticate against an OpenGemini server that requires authentication. The https field controls whether to use the HTTPS protocol. 

- [x] Test A
- [ ] Test B
- [ ] Test cases to be added
- [ ] No code

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by xiangyu5632** [comment]
Some test cases need to be added please  @yehong-z

**Comment by xiangyu5632** [comment]
Please check the checklist in the PR. @yehong-z

File: lib/config/monitor.go [hunk 1]
```diff
@@ -136,6 +136,9 @@ type MonitorReport struct {
 	Database   string        `toml:"database"`
 	Rp         string        `toml:"rp"`
 	RpDuration toml.Duration `toml:"rp-duration"`
+	UserName   string        `toml:"username"`
+	Password   string        `toml:"password"`
+	UseHttps   bool          `toml:"https"`
```
Comment (shilinlee) [inline lib/config/monitor.go:141]: HTTPSEnabled        bool `toml:"https-enabled"`

File: lib/config/monitor.go [hunk 2]
```diff
@@ -136,6 +136,9 @@ type MonitorReport struct {
 	Database   string        `toml:"database"`
 	Rp         string        `toml:"rp"`
 	RpDuration toml.Duration `toml:"rp-duration"`
+	UserName   string        `toml:"username"`
```
Comment (shilinlee) [inline lib/config/monitor.go:139]: UserName -> Username

File: app/ts-monitor/collector/report.go [hunk 3]
```diff
@@ -102,13 +102,14 @@ type ReportJob struct {
 	reportStat *ReportStat
 }
 
-func NewReportJob(addr, database, storeRP string, rpDuration time.Duration, gzipped, compress bool, logger *logger.Logger, errLogHistory string) *ReportJob {
+func NewReportJob(addr, database, storeRP, username, password string, useHttps bool, rpDuration time.Duration, gzipped, compress bool, logger *logger.Logger, errLogHistory string) *ReportJob {
+	protocol := map[bool]string{true: "https", false: "http"}[useHttps]
```
Comment (shilinlee) [inline app/ts-monitor/collector/report.go:106]: use this simple way to specify the protocol

``` go
protocol := "http"
if useHttps {
    protocol = "https"
}
```

File: app/ts-monitor/collector/report.go [hunk 4]
```diff
@@ -102,13 +102,14 @@ type ReportJob struct {
...
 	mr := &ReportJob{
 		storeDatabase: database,
 		storeRP:       storeRP,
 		storeDuration: rpDuration,
-		writeUrl:      fmt.Sprintf("http://%s/write?db=%s&rp=%s", addr, database, storeRP),
-		queryUrl:      fmt.Sprintf("http://%s/query", addr),
+		writeUrl:      fmt.Sprintf("%s://%s/write?db=%s&rp=%s&u=%s&p=%s", protocol, addr, database, storeRP, username, password),
+		queryUrl:      fmt.Sprintf("%s://%s/query?u=%s&p=%s", protocol, addr, username, password),
```
Comment (shilinlee) [inline app/ts-monitor/collector/report.go:112]: `queryUrl` should not be set protocol and auth, it may not be the same source as `writeUrl`.

**Review by shilinlee — CHANGES_REQUESTED** [review]

File: config/monitor.conf [hunk 5]
```diff
@@ -35,6 +35,9 @@
   database = "monitor"
   rp = "autogen"
   rp-duration = "168h"
+  # username = ""
+  # password = ""
+  # https-enable = true
```
Comment (shilinlee) [inline config/monitor.conf:40]: please change default value to false

**Review by shilinlee — CHANGES_REQUESTED** [review]

File: config/monitor.conf [hunk 6]
```diff
@@ -35,6 +35,9 @@
   database = "monitor"
   rp = "autogen"
   rp-duration = "168h"
+  # username = ""
+  # password = ""
+  # https-enable = false
```
Comment (shilinlee) [inline config/monitor.conf:40]: please change default value to false

**Review by shilinlee — CHANGES_REQUESTED** [review]

**Review by shilinlee — APPROVED** [review]

**Pull request merged**